### PR TITLE
Add CLUSTER_NAME environment variable check for citizen update case state

### PR DESCRIPTION
**Pull request by lara-mpt**
### Change description ###
Add CLUSTER_NAME environment variable check for citizen update case state

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/divorce/citizen/event/CitizenUpdateCaseStateAat.java [hunk 1]
```diff
@@ -54,9 +56,11 @@ public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuild
...
     }
 
-    public boolean isEnvironmentAat() {
+    public boolean isEnvironmentAatOrPreview() {
         String environment = System.getenv().getOrDefault("ENVIRONMENT", null);
-        return null != environment && environment.equalsIgnoreCase(ENVIRONMENT_AAT);
+        String clusterName =  System.getenv().getOrDefault("CLUSTER_NAME", null);
+        return null != environment && (environment.equalsIgnoreCase(ENVIRONMENT_AAT) || environment.equalsIgnoreCase(ENVIRONMENT_PREVIEW))
+            || clusterName != null && (clusterName.contains(ENVIRONMENT_AAT) || clusterName.contains(ENVIRONMENT_PREVIEW));
     }
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/divorce/citizen/event/CitizenUpdateCaseStateAat.java:64]: wouldn't be easier to check if environment is not PROD ?

Comment (lara-mpt) [reply src/main/java/uk/gov/hmcts/divorce/citizen/event/CitizenUpdateCaseStateAat.java:64]: yeah, if that's fine to allow its use on demo etc..

**Comment by linusnorton** [comment]
@lara-mpt you will need this at build time so you will also need to add that env var to the Jenkinsfile

File: src/main/java/uk/gov/hmcts/divorce/citizen/event/CitizenUpdateCaseStateAat.java [hunk 2]
```diff
@@ -54,9 +52,8 @@ public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuild
             .build();
     }
 
-    public boolean isEnvironmentAat() {
-        String environment = System.getenv().getOrDefault("ENVIRONMENT", null);
-        return null != environment && environment.equalsIgnoreCase(ENVIRONMENT_AAT);
+    public boolean isCitizenUpdateCaseStateEnabled() {
+        String citizenUpdateCaseStateEnabled = System.getenv().getOrDefault("CITIZEN_UPDATE_CASE_STATE_ENABLED", null);
+        return null != citizenUpdateCaseStateEnabled && citizenUpdateCaseStateEnabled.equalsIgnoreCase("true");
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/divorce/citizen/event/CitizenUpdateCaseStateAat.java:57]: could we just do 
```
return Boolean.valueOf(System.getenv().get("CITIZEN_UPDATE_CASE_STATE_ENABLED"))
```

**Review by nitinprabhu — APPROVED** [review]

**Pull request merged**